### PR TITLE
Clarify build timeouts for mac (#3810)

### PR DESCRIPTION
**Pull request by winterchord**
Mac build timeouts are hard set to 50 minutes.

The commit introduces the terms, hard timeout and idle-output timeouts as the
two types of possible timeouts for a job.

This fixes https://github.com/travis-ci/travis-ci/issues/3810

### CODE REVIEW

**Comment by winterchord** [comment]
@BanzaiMan When you have a moment, can you review this documentation update?

**Comment by plaindocs** [comment]
I'd reword this to something like

> It is very common for test suites or build scripts to hang. Travis CI has specific time limits for each 
> job, and will stop the build and and add an error message to the build log in the following situations:
>
> * A job takes longer than 50 minutes on travis-ci.org
> * A job takes longer than 120 minutes on travis-ci.com
> * A job takes longer than 50 minutes on OSX infrastructure on travis-ci.org or travis-ci.com
> * A job produces no log output for 10 minutes

Add the Hard / Idle timeouts descriptors if they are important.

**Comment by MariadeAnton** [comment]
@winterchord Thank you so much for the PR! Could you update it with @plaindocs's recommendations?

**Comment by winterchord** [comment]
@MariadeAnton Sorry for the delay. Amended and force pushed the suggested changes from @plaindocs.

**Comment by plaindocs** [comment]
Thanks for the PR and subsequent change.
